t/socket-tcp-getoption.t

```diff
@@ -0,0 +1,288 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use lib '.';
+use t::TestCore;
+
+repeat_each(2);
+
+plan tests => repeat_each() * (blocks() * 2 + 7);
+
+#worker_connections(1024);
+#no_diff();
+no_long_string();
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: no parameters.
+--- config
+    set $port $TEST_NGINX_SERVER_PORT;
+
+    location /t {
+        content_by_lua_block {
+            require "resty.core.socket"
+
+            local port = ngx.var.port
+            local sock = ngx.socket.tcp()
+            local ok, err = sock:connect("127.0.0.1", port)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            ok, err = sock:getoption()
+            if not ok then
+                ngx.say("getoption failed: ", err)
+                return
+            end
+
+            sock:close()
+        }
+    }
+--- request
+GET /t
+--- response_body
+getoption failed: missing the "option" argument
+--- no_error_log
+[error]
+
+
+
+=== TEST 2: unsuppotrted option name.
+--- config
+    set $port $TEST_NGINX_SERVER_PORT;
+
+    location /t {
+        content_by_lua_block {
+            require "resty.core.socket"
+
+            local port = ngx.var.port
+            local sock = ngx.socket.tcp()
+            local ok, err = sock:connect("127.0.0.1", port)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            local sndbuf, err = sock:getoption("abc")
+            if not sndbuf then
+                ngx.say("getoption abc failed: ", err)
+                return
+            end
+
+            sock:close()
+        }
+    }
+--- request
+GET /t
+--- response_body
+getoption abc failed: unsupported option abc
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: getoption before calling connect.
+--- config
+    set $port $TEST_NGINX_SERVER_PORT;
+
+    location /t {
+        content_by_lua_block {
+            require "resty.core.socket"
+
+            local sock = ngx.socket.tcp()
+            local sndbuf, err = sock:getoption("so_sndbuf")
```

Comment:
Using the wrong option name here? It's not testing what you meant to test.

Reply:
accept, "so_sndbuf" => "sndbuf"

Reply:
Still not changed here?